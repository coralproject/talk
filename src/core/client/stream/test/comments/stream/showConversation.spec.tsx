import { ReactTestRenderer } from "react-test-renderer";
import sinon from "sinon";

import {
  act,
  createSinonStub,
  waitForElement,
  within,
} from "coral-framework/testHelpers";

import { comments, settings, storyWithDeepestReplies } from "../../fixtures";
import create from "./create";

let testRenderer: ReactTestRenderer;
beforeEach(() => {
  const resolvers = {
    Query: {
      story: createSinonStub(
        s => s.throws(),
        s => s.returns(storyWithDeepestReplies)
      ),
      stream: createSinonStub(
        s => s.throws(),
        s => s.returns(storyWithDeepestReplies)
      ),
      comment: createSinonStub(
        s => s.throws(),
        s =>
          s
            .withArgs(undefined, { id: "comment-with-deepest-replies-3" })
            .returns({
              ...comments[0],
              id: "comment-with-deepest-replies-3",
            })
      ),
      settings: sinon.stub().returns(settings),
    },
  };

  ({ testRenderer } = create({
    // Set this to true, to see graphql responses.
    logNetwork: false,
    resolvers,
    initLocalState: localRecord => {
      localRecord.setValue(storyWithDeepestReplies.id, "storyID");
    },
  }));
});

it("renders deepest comment with link", async () => {
  const streamLog = await waitForElement(() =>
    within(testRenderer.root).getByTestID("comments-allComments-log")
  );
  const deepestReply = within(streamLog).getByTestID(
    "comment-comment-with-deepest-replies-3"
  );
  expect(within(deepestReply).toJSON()).toMatchSnapshot();
});

it("shows conversation", async () => {
  const mockEvent = {
    preventDefault: sinon.mock().once(),
  };
  const streamLog = await waitForElement(() =>
    within(testRenderer.root).getByTestID("comments-allComments-log")
  );
  await act(async () => {
    within(streamLog)
      .getByText("Read More of this Conversation", { exact: false })
      .props.onClick(mockEvent);

    await waitForElement(() =>
      within(testRenderer.root).getByText("SINGLE CONVERSATION")
    );
  });
});
