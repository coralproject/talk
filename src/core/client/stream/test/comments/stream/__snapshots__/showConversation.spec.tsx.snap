// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders deepest comment with link 1`] = `
<div
  className="coral coral-comment coral coral-reacted-0"
  data-testid="comment-comment-with-deepest-replies-3"
>
  <div
    className="Box-root HorizontalGutter-root HorizontalGutter-full"
  >
    <div
      className="Indent-root"
    >
      <div
        className="Indent-level3 coral coral-indent coral-indent-3"
      >
        <div
          className="Box-root HorizontalGutter-root Comment-root HorizontalGutter-half"
          role="article"
        >
          <div
            className="Box-root Flex-root coral coral-comment-topBar Flex-flex Flex-justifySpaceBetween Flex-directionRow"
          >
            <div
              className="Box-root Flex-root Flex-flex Flex-halfItemGutter Flex-alignBaseline Flex-directionColumn gutter"
            >
              <div
                className="Box-root Flex-root Flex-flex Flex-halfItemGutter Flex-alignCenter Flex-directionRow gutter"
              >
                <div
                  className="Popover-root"
                >
                  <button
                    aria-controls="username-popover-comment-with-deepest-replies-3"
                    className="BaseButton-root coral coral-username coral-comment-username"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    type="button"
                  >
                    <span
                      className="Username-root"
                    >
                      Markus
                    </span>
                  </button>
                  <div
                    aria-hidden={true}
                    aria-labelledby="username-popover-comment-with-deepest-replies-3-ariainfo"
                    id="username-popover-comment-with-deepest-replies-3"
                    role="dialog"
                  >
                    <div
                      className="AriaInfo-root"
                      id="username-popover-comment-with-deepest-replies-3-ariainfo"
                    >
                      A popover with more user information
                    </div>
                  </div>
                </div>
                <div
                  className="Box-root Flex-root Flex-flex Flex-alignCenter"
                />
              </div>
              <div
                className="Box-root Flex-root Flex-flex Flex-itemGutter Flex-alignBaseline Flex-directionRow gutter"
              >
                <button
                  className="BaseButton-root Timestamp-root"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <time
                    className="RelativeTime-root Timestamp-text Comment-timestamp coral coral-timestamp coral-comment-timestamp"
                    dateTime="2018-07-06T18:24:00.000Z"
                    title="2018-07-06T18:24:00.000Z"
                  >
                    2018-07-06T18:24:00.000Z
                  </time>
                </button>
              </div>
            </div>
            <div>
              <div
                className="Box-root Flex-root Flex-flex Flex-itemGutter Flex-alignCenter gutter"
              />
            </div>
          </div>
          <div
            className="Comment-subBar"
          >
            <div
              className="Box-root Flex-root coral coral-comment-inReplyTo Flex-flex Flex-alignCenter"
            >
              <i
                aria-hidden="true"
                className="Icon-root Icon-sm InReplyTo-icon"
              >
                reply
              </i>
              <span>
                Â 
              </span>
              <span
                className="Box-root Typography-root Typography-timestamp Typography-colorTextPrimary InReplyTo-inReplyTo coral coral-comment-inReplyToText"
              >
                In reply to 
                <span
                  className="Box-root Typography-root Typography-heading4 Typography-colorTextDark InReplyTo-username coral coral-comment-inReplyToUsername"
                >
                  Markus
                </span>
              </span>
            </div>
          </div>
          <div
            className="Box-root HorizontalGutter-root HorizontalGutter-oneAndAHalf"
          >
            <div
              className="HTMLContent-root coral coral-content coral-comment-content"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "body 3",
                }
              }
            />
            <div
              className="Box-root Flex-root coral coral-comment-actionBar Flex-flex Flex-justifySpaceBetween"
            >
              <div
                className="Box-root Flex-root CommentContainer-actionBar Flex-flex Flex-halfItemGutter Flex-justifySpaceBetween Flex-alignCenter Flex-directionRow gutter"
              >
                <button
                  aria-label="Respect"
                  className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantText Button-uppercase CommentContainer-actionButton coral coral-reactButton coral-comment-reactButton ReactionButton-button"
                  data-color="mono"
                  data-testid="comment-reaction-button"
                  data-variant="text"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="Icon-root Icon-sm"
                  >
                    thumb_up
                  </i>
                </button>
                <button
                  aria-label="Reply"
                  className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantText Button-uppercase CommentContainer-actionButton coral coral-comment-replyButton"
                  data-active={false}
                  data-color="mono"
                  data-testid="comment-reply-button"
                  data-variant="text"
                  disabled={false}
                  id="comments-commentContainer-replyButton-comment-with-deepest-replies-3"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="Icon-root Icon-sm"
                  >
                    reply
                  </i>
                </button>
                <div
                  className="Popover-root"
                >
                  <button
                    aria-controls="permalink-popover-comment-with-deepest-replies-3"
                    aria-label="Share"
                    className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantText Button-uppercase CommentContainer-actionButton coral coral-comment-shareButton"
                    data-active={false}
                    data-color="mono"
                    data-variant="text"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    type="button"
                  >
                    <i
                      aria-hidden="true"
                      className="Icon-root Icon-sm"
                    >
                      share
                    </i>
                  </button>
                  <div
                    aria-hidden={true}
                    aria-labelledby="permalink-popover-comment-with-deepest-replies-3-ariainfo"
                    id="permalink-popover-comment-with-deepest-replies-3"
                    role="dialog"
                  >
                    <div
                      className="AriaInfo-root"
                      id="permalink-popover-comment-with-deepest-replies-3-ariainfo"
                    >
                      A dialog showing a permalink to the comment
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="Box-root Flex-root Flex-flex Flex-halfItemGutter Flex-justifySpaceBetween Flex-alignCenter Flex-directionRow gutter"
              >
                <button
                  aria-label="Report"
                  className="BaseButton-root Button-root Button-sizeRegular Button-colorMono Button-variantText Button-uppercase coral coral-reportButton coral-comment-reportButton"
                  data-color="mono"
                  data-testid="comment-report-button"
                  data-variant="text"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  onTouchEnd={[Function]}
                  type="button"
                >
                  <i
                    aria-hidden="true"
                    className="Icon-root Icon-sm"
                  >
                    flag
                  </i>
                </button>
              </div>
            </div>
            <a
              className="BaseButton-root Button-root Button-sizeRegular Button-colorRegular ShowConversationLink-colorRegular Button-variantTextUnderlined Button-uppercase ShowConversationLink-conversationLink coral coral-comment-readMoreOfConveration"
              data-color="regular"
              data-variant="textUnderlined"
              href="http://localhost/stories/story-with-replies?commentID=comment-with-deepest-replies-3"
              id="comments-commentContainer-showConversation-comment-with-deepest-replies-3"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              onTouchEnd={[Function]}
              target="_parent"
              type="button"
            >
              Read More of this Conversation &gt;
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
